sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -1,375 +0,0 @@
-
-
-@ignore_warnings
-def _check_statistics(X, X_true,
-                      strategy, statistics, missing_values):
-    """Utility function for testing imputation for a given strategy.
-
-    Test:
-        - along the two axes
-        - with dense and sparse arrays
-
-    Check that:
-        - the statistics (mean, median, mode) are correct
-        - the missing values are imputed correctly"""
-
-    err_msg = "Parameters: strategy = %s, missing_values = %s, " \
-              "axis = {0}, sparse = {1}" % (strategy, missing_values)
-
-    assert_ae = assert_array_equal
-    if X.dtype.kind == 'f' or X_true.dtype.kind == 'f':
-        assert_ae = assert_array_almost_equal
-
-    # Normal matrix, axis = 0
-    imputer = Imputer(missing_values, strategy=strategy, axis=0)
-    X_trans = imputer.fit(X).transform(X.copy())
-    assert_ae(imputer.statistics_, statistics,
-              err_msg=err_msg.format(0, False))
-    assert_ae(X_trans, X_true, err_msg=err_msg.format(0, False))
-
-    # Normal matrix, axis = 1
-    imputer = Imputer(missing_values, strategy=strategy, axis=1)
-    imputer.fit(X.transpose())
-    if np.isnan(statistics).any():
-        assert_raises(ValueError, imputer.transform, X.copy().transpose())
-    else:
-        X_trans = imputer.transform(X.copy().transpose())
-        assert_ae(X_trans, X_true.transpose(),
-                  err_msg=err_msg.format(1, False))
-
-    # Sparse matrix, axis = 0
-    imputer = Imputer(missing_values, strategy=strategy, axis=0)
-    imputer.fit(sparse.csc_matrix(X))
-    X_trans = imputer.transform(sparse.csc_matrix(X.copy()))
-
-    if sparse.issparse(X_trans):
-        X_trans = X_trans.toarray()
-
-    assert_ae(imputer.statistics_, statistics,
-              err_msg=err_msg.format(0, True))
-    assert_ae(X_trans, X_true, err_msg=err_msg.format(0, True))
-
-    # Sparse matrix, axis = 1
-    imputer = Imputer(missing_values, strategy=strategy, axis=1)
-    imputer.fit(sparse.csc_matrix(X.transpose()))
-    if np.isnan(statistics).any():
-        assert_raises(ValueError, imputer.transform,
-                      sparse.csc_matrix(X.copy().transpose()))
-    else:
-        X_trans = imputer.transform(sparse.csc_matrix(X.copy().transpose()))
-
-        if sparse.issparse(X_trans):
-            X_trans = X_trans.toarray()
-
-        assert_ae(X_trans, X_true.transpose(),
-                  err_msg=err_msg.format(1, True))
-
-
-@ignore_warnings
-def test_imputation_shape():
-    # Verify the shapes of the imputed matrix for different strategies.
-    X = np.random.randn(10, 2)
-    X[::2] = np.nan
-
-    for strategy in ['mean', 'median', 'most_frequent']:
-        imputer = Imputer(strategy=strategy)
-        X_imputed = imputer.fit_transform(X)
-        assert_equal(X_imputed.shape, (10, 2))
-        X_imputed = imputer.fit_transform(sparse.csr_matrix(X))
-        assert_equal(X_imputed.shape, (10, 2))
-
-
-@ignore_warnings
-def test_imputation_mean_median_only_zero():
```

Comment:
It doesn't look like this test is present in `sklearn/tests/test_impute.py`. Do we want to copy it there? Unless it was deemed redundant ?